### PR TITLE
V balajium/holdand un hold changes

### PR DESCRIPTION
**Pull request by BalajiUmmadisetti**
# Description

US: https://skype.visualstudio.com/SPOOL/_workitems/edit/3621290
while testing SDK getting revapi - java.method.numberOfParametersChanged issue, so didn't change existing constructor , just added new constructor method with 3 params.
Added CallParticipant constructor with 3 params
public CallParticipant(CommunicationIdentifier identifier, Boolean isMuted, Boolean isOnHold).

Tested using Contoso Bank Outbound call quick start. 
Tested Scenarios:
1.hold
2.holdWithResponse
3.unhold
4.unholdWithResponse
5.CallParticipant with isOnHold option.

Swagger spec: https://github.com/Azure/azure-rest-api-specs/blob/77d25dd8426c4ba1619d15582a8c9d9b2f6890e8/specification/communication/data-plane/CallAutomation/readme.md

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+     * Operation context.
+     */
+    private String operationContext;
+
+    /**
+     * Create a new HoldOptions object.
+     * @param targetParticipant Participant to be put on hold.
+     * @param playSourceInfo Audio to be played while on hold.
+     */
+    public HoldOptions(CommunicationIdentifier targetParticipant, PlaySource playSourceInfo) {
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java:33]: `playSourceInfo` is optional. Please remove it from the constructor.

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaAsyncUnitTests.java [hunk 2]
```diff
@@ -397,6 +398,32 @@ public void stopHoldMusicWithResponseTest() {
             .consumeNextWith(response -> assertEquals(200, response.getStatusCode()))
             .verifyComplete();
     }
+    
+    @Test
+    public void holdWithResponseTest() {
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaAsyncUnitTests.java:403]: Please add a test with no prompt.

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaUnitTests.java [hunk 3]
```diff
@@ -214,6 +214,33 @@ public void stopHoldMusicWithResponseTest() {
...
+    @Test
+    public void holdWithResponseTest() {
+        CallConnection callConnection =
+            CallAutomationUnitTestBase.getCallConnection(new ArrayList<>(
+                Collections.singletonList(new AbstractMap.SimpleEntry<>("", 200)))
+            );
+        callMedia = callConnection.getCallMedia();
+        HoldOptions options = new HoldOptions(
+            new CommunicationUserIdentifier("id"),
+            new TextSource().setText("audio to play"));
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaUnitTests.java:226]: Please do one test with no music.

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+// Licensed under the MIT License.
+
+package com.azure.communication.callautomation.models;
+
+import com.azure.communication.common.CommunicationIdentifier;
+
+/**
+ * Options for the Hold operation.
+ */
+public class HoldOptions {
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java:11]: this is missing the optional call back URI.

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java (hunk 1 above)
Comment (BalajiUmmadisetti) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java]: changed playSourceInfo  to optional

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaAsyncUnitTests.java (hunk 2 above)
Comment (BalajiUmmadisetti) [reply sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaAsyncUnitTests.java:403]: added a test with no prompt.

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaUnitTests.java (hunk 3 above)
Comment (BalajiUmmadisetti) [reply sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaUnitTests.java]: added a test with no prompt.

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java (hunk 4 above)
Comment (BalajiUmmadisetti) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java:11]: added operationCallbackUri

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/implementation/models/CallParticipantInternal.java [hunk 5]
```diff
@@ -61,4 +67,24 @@ public CallParticipantInternal setIsMuted(Boolean isMuted) {
         this.isMuted = isMuted;
         return this;
     }
+
+    /**
+     * Get the isOnHold property: Is participant on hold.
+     *
+     * @return the isOnHold value.
+     */
+    public Boolean isOnHold() {
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/implementation/models/CallParticipantInternal.java:76]: Please update `CallParticipant` as well: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CallParticipant.java

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/implementation/models/HoldFailed.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.communication.callautomation.implementation.models;
+
+import com.azure.core.annotation.Immutable;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/** The HoldFailed model. */
+@Immutable
+public final class HoldFailed {
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/implementation/models/HoldFailed.java:12]: Please, add these event in the Models/Events. 
As well, update the Event Parser: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallAutomationEventParser.java

Comment (BalajiUmmadisetti) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/implementation/models/HoldFailed.java:12]: added event in the Models/Events and updated the Event Parser

**Review by Paresh-Arvind-Patil — APPROVED** [review]

**Pull request merged**